troubleshooting/faq.md

```diff
@@ -161,6 +162,33 @@ So it’s important to understand the different storage driver options available
 
 We recommend that you use the `overlay` driver, rather than `devicemapper`; for help configuring this [please see the following tutorial](https://docs.docker.com/engine/userguide/storagedriver/overlayfs-driver/#configure-docker-with-the-overlayoverlay2-storage-driver)
 
+## What should I do with git dependencies on closed networks?
+
+If you are installing a package that has a git URL as a dependency, npm will fetch that package from the remote git repository itself. npm supports both git and HTTP URL formats (see https://docs.npmjs.com/files/package.json#dependencies), so we will need to work around this in our private registry.
+
+```
+"dependencies" : {
+  "name1" : "git://github.com/user/project.git#commit-ish",
+  "name2" : "git://github.com/user/project.git#commit-ish"
+}
+```
+When you are installing the package in the closed network where a dependency references a git url rather than a package name, you will be unable to access due to no internet connectivity.
+
+In this situation, one approach you can take is to download a tarball version of the package and republish it to your private registry.
+
+You can download a tarball version of the package from GitHub using
+```
+ wget -L https://github.com/user-or-org/repo/archive/master.tar.gz
+
+```
+Replace `user-or-org` and `repo` accordingly.
+
+To publish the tarball to private registry simply follow the steps listed below:
+
+*  tar -xvzf package.tar.gz
+*  cd package
+*  npm publish --registry=http://your-private-registry:8080
```

Comment:
thinking about it, we should probably add one more section that describes how to use the new dependency;

People will need to switch the packages referencing the git URL, to reference the dependency that is now published to their private registry.

Perhaps we can split this into two sections:

`publishing the git dependency to your private repo`

and

`updating packages to reference the new dependency`

We should note that, since it will often be a dependency of a dependency that has the git dependency, it will likely be necessary to both publish the git dependency and the dependency requiring it. A good best practice is to publish these new versions with a scope (@mycompany/request);

we have a tool for simplifying this process: https://www.npmjs.com/package/scope-it